nbgrader/alembic/versions/e43177bfe90b_added_course_table_and_relationships.py

```diff
@@ -0,0 +1,47 @@
+"""Added course table and relationships
+
+Revision ID: e43177bfe90b
+Revises: 282d30a0218e
+Create Date: 2019-05-30 09:39:31.881296
+
+"""
+from alembic import op
+import sqlalchemy as sa
+from nbgrader.api import Course
+
+# revision identifiers, used by Alembic.
+revision = 'e43177bfe90b'
+down_revision = '282d30a0218e'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    """
+    This migrations adds a course column to the assignment table
+    and a matching foreign key into the course table
+    """
+
+    # If the course table is empty, create a default course
+    connection = op.get_bind()
+    res = connection.execute("select id from course")
+    results = res.fetchall()
+    default_course = "default_course"
+
+    if len(results) == 0:
+        connection.execute(
+            "INSERT INTO course (id) VALUES ('{}')".format(default_course))
+
+    with op.batch_alter_table("assignment") as batch_op:
+
+        batch_op.add_column(sa.Column(
+            'course_id', sa.VARCHAR(128), nullable=False, default=default_course))
+        batch_op.create_foreign_key(
```

Comment:
Do you know if `fk_course_assignment` is the same name as the foreign key created when the database is created from scratch? If not I'm just wondering if that could potentially cause problems later down the line...

Reply:
In Sqlite, the foreign key constraint is not named. The migration seems to work in both directions, up and down, but I don't know whether I've considered all cases.

Reply:
Ok, sounds good! Thanks for clarifying. I think it's fine if the constraint normally doesn't have an name.